static/resources/js/components/details/retrieve-projects.directive.js

```diff
@@ -62,33 +66,49 @@
       }
     });
     
-    function getProjectSuccess(data, status) {
-      vm.projects = data || [];
-      if(vm.projects.length == 0 && vm.isPublic === 0){
-        $window.location.href = '/project';  
+    function getProjectSuccess(response) {
+      
+      var partialProjects = response.data || []; 
+      for(var i in partialProjects) {
+        vm.projects.push(partialProjects[i]);
       }
-                
-      if(getParameterByName('project_id', $location.absUrl())){
-        for(var i in vm.projects) {
-          var project = vm.projects[i];
-          if(project['project_id'] == getParameterByName('project_id', $location.absUrl())) {
-            vm.selectedProject = project;
-            break;
-          }
-        } 
+      
+      var link = response.headers("Link") || '';
+      var pattern = /^<\/api\/projects\?is_public=(\d+)&page\=(\d+)&page_size=(\d+)\&project_name=>; rel=\"(\w+)\"$/;
```

Comment:
I think you should only need to get the url and trigger a request, you don't need to care what the url of "next page" is